### PR TITLE
Fix #1605: Add more fixes to #1653

### LINKED ISSUE #1605: Add Challenge preview mode for the challenge hosts
## Description
The challenge host must be able to view the challenge in preview mode before publishing the challenge. Also, they must be able to make edits and make submissions using their account.

## Proposed Approach

- When the challenge `is_active` flag is false, challenge host can only see the challenge and make submissions and see their entry on the leaderboard for testing purposes.

- When the challenge’s `is_active` flag is true, challenge hosts and others can see the challenge and make submissions but their submissions won’t go on leaderboard.

## Deliverables
- [ ] Update the backend API for showing the challenge to challenge host only.
- [ ] Update the leaderboard API that will exclude the submissions from the challenge host team.

### PR DESCRIPTION
**Pull request by deshraj**
This PR improves #1653 and fixes #1605. 

@isht3 see the screenshots below. Also, please do look at the code. You will see that what all things you could have improves. :) 

screen shot 2018-04-19 at 9 49 09 pm
screen shot 2018-04-19 at 9 49 24 pm

### CODE REVIEW

**Comment by RishabhJain2018** [comment]
@deshraj The page appears to be different on my system. Please take a look!
screen shot 2018-04-20 at 7 58 35 am

**Comment by deshraj** [comment]
@RishabhJain2018 I think you are testing something else since the sidebar in your screenshot is different than that of what I have. Can you please check that?

File: apps/jobs/views.py [hunk 1]
```diff
@@ -214,11 +214,18 @@ def leaderboard(request, challenge_phase_split_id):
         response_data = {'error': 'Sorry, Default filtering key not found in leaderboard schema!'}
         return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
 
+    # Exclude the submissions done by members of the host team
+    # while populating leaderboard
+    challenge_obj = challenge_phase_split.challenge_phase.challenge
+    challenge_hosts_emails = challenge_obj.creator.get_all_challenge_host_email()
+    leaderboard_data = LeaderboardData.objects.exclude(
```
[the remark this answers is not in the document]
Comment (deshraj) [reply apps/jobs/views.py]: No worries. I am glad that you know how to do it now. :)

**Comment by deshraj** [comment]
@RishabhJain2018 it could be the case that while resolving the merge conflicts, I might have messed up things and that's why you are seeing these kind of issues on your system. I will check again on my system and see if I face the same issue.

**Comment by RishabhJain2018** [comment]
@deshraj Now it works on my system .

**Comment by RishabhJain2018** [comment]
@deshraj It works perfectly on my system. I have added some minor fixes in it. Please take a look at them.

**Pull request merged**